File: 167-two-sum-ii/solution.py
from typing import List


class Solution:
  def twoSum(self, numbers: List[int], target: int) -> List[int]:
    i, j = 0, len(numbers) - 1
    while i < j:
      needed = target - numbers[i]
      while numbers[j] > needed:
        j -= 1

      if numbers[j] == needed:
        break

      i += 1

    return [i + 1, j + 1]

File: 167-two-sum-ii/solution.cpp
#include <vector>

class Solution
{
public:
  std::vector<int> twoSum(std::vector<int> &numbers, int target)
  {
    int i = 0;
    int j = numbers.size() - 1;
    while (i < j)
    {
      const int needed = target - numbers[i];
      while (numbers[j] > needed)
      {
        --j;
      }

      if (numbers[j] == needed)
      {
        break;
      }

      ++i;
    }

    return {i + 1, j + 1};
  }
};